### PR TITLE
Remove finish_observing 

### PR DESCRIPTION
**Pull request by danjampro**
Removes outdated code from `observatory` and `observing` state.

### CODE REVIEW

File: src/huntsman/pocs/states/huntsman/observing.py [hunk 1]
```diff
@@ -16,19 +16,15 @@ def on_enter(event_data):
 
         wait_time = 0.
         while not all([event.is_set() for event in camera_events.values()]):
```
Comment (JAAlvarado-Montes) [inline src/huntsman/pocs/states/huntsman/observing.py:18]: Isn't it better using `panoptes.pocs.time.wait_for_events` here?

**Review by JAAlvarado-Montes — COMMENTED** [review]
LGTM. Just a small comment about using `wait_for_events`.

File: src/huntsman/pocs/states/huntsman/observing.py (hunk 1 above)
Comment (danjampro) [reply src/huntsman/pocs/states/huntsman/observing.py:18]: @JAAlvarado-Montes Yes!

Comment (danjampro) [reply src/huntsman/pocs/states/huntsman/observing.py:18]: Although I think we can leave this to a more generic cleanup of the `observe` function, which IMHO should/could be blocking.

Comment (JAAlvarado-Montes) [reply src/huntsman/pocs/states/huntsman/observing.py:18]: OK, sounds good!

**Review by AnthonyHorton — APPROVED** [review]
LGTM.

**Pull request merged**